daemon/cmd/kube_proxy_replacement.go

```diff
@@ -347,6 +345,59 @@ func initKubeProxyReplacementOptions() (strict bool) {
 	return
 }
 
+func probeCgroupSupportTCP(strict, ipv4 bool) {
+	var scopedLog *logrus.Entry
+	var disableHostServicesTCP bool
+
+	if ipv4 {
+		if err := bpf.TestDummyProg(bpf.ProgTypeCgroupSockAddr, bpf.BPF_CGROUP_INET4_CONNECT); err != nil {
+			scopedLog = log.WithError(err)
+			disableHostServicesTCP = true
+		}
+	} else {
+		if err := bpf.TestDummyProg(bpf.ProgTypeCgroupSockAddr, bpf.BPF_CGROUP_INET6_CONNECT); err != nil {
+			scopedLog = log.WithError(err)
+			disableHostServicesTCP = true
```

Comment:
These two lines can be moved out of the if/else block for IPv4/IPv6 and merged with the following code, such that we don't need `disableHostServicesTCP` anymore.